src/Elliptic/Systems/Poisson/FirstOrderSystem.hpp

```diff
@@ -84,18 +87,18 @@ struct FirstOrderSystem {
   using fields_tag =
       ::Tags::Variables<tmpl::append<primal_fields, auxiliary_fields>>;
 
-  using fluxes =
-      tmpl::conditional_t<BackgroundGeometry == Geometry::Euclidean,
-                          EuclideanFluxes<Dim>, NonEuclideanFluxes<Dim>>;
-  using sources = Sources;
+  using fluxes = Fluxes<Dim, BackgroundGeometry>;
+  using sources = Sources<Dim, BackgroundGeometry>;
 
   // The tag of the operator to compute magnitudes on the manifold, e.g. to
   // normalize vectors on the faces of an element
+  using inv_metric_tag = tmpl::conditional_t<
+      BackgroundGeometry == Geometry::Euclidean, void,
+      gr::Tags::InverseSpatialMetric<Dim, Frame::Inertial, DataVector>>;
```

Comment:
These tags are tightly coupled to the choice of Geometry.  Another option would be to turn Geometry::Euclidean and Geometry::NonEuclidean into structs having members 'inv_metric_tag' and 'magnitude_tag'

Reply:
That's certainly possible, though I think it's cleaner to have the system struct deal with selecting tags and have the enum only provide the cases.